src/pci.rs

```diff
@@ -96,6 +118,22 @@ impl PciConfig {
 
         self.read_at(addr)
     }
+
+    #[allow(unused)]
+    fn write(&mut self, bus: u8, device: u8, func: u8, offset: u8, value: u32) {
```

Comment:
Well, segments are tricky. They can only be found through a PCI ECAM region. Finding that ECAM region is easy (and already done) on FDT platforms. On x86 it would need to involved looking it up in the MCFG table (and currently there is no ACPI table passing in this code).

The use of multiple segments is definitely not mainstream and booting from the non-zero segment seems very niche.

Reply:
Fair enough!